azure-kusto-data/azure/kusto/data/request.py

```diff
@@ -1,83 +1,194 @@
-import requests
+"""A module to make a Kusto request."""
+
 import uuid
+from enum import Enum, unique
+import requests
 
 from .security import _AadHelper
 from .exceptions import KustoServiceError
 from ._response import _KustoResponseDataSetV1, _KustoResponseDataSetV2
 from ._version import VERSION
 
 
-class KustoClient(object):
+class KustoConnectionStringBuilder(object):
+    """Parses Kusto conenction strings.
+    For usages, check out the sample at:
+        https://github.com/Azure/azure-kusto-python/blob/master/azure-kusto-data/tests/sample.py
     """
-    Kusto client for Python.
 
-    KustoClient works with both 2.x and 3.x flavors of Python. All primitive types are supported.
-    KustoClient takes care of ADAL authentication, parsing response and giving you typed result set.
+    @unique
+    class ValidKeywords(Enum):

```

Comment:
Only a logic reason. Those are the valid keywords to build a Kusto connection string.

Reply:
One thing to test is that you can put these values into a dict and round-trip it using `pickle`. Sometimes it can't handle certain nesting, though I think this one will be fine.